modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt

```diff
@@ -1108,4 +995,148 @@ fun <A> IOOf<A>.handleError(f: (Throwable) -> A): IO<A> =
  * @see handleErrorWith for a version that can resolve the error using an effect
  */
 fun <A> IOOf<A>.handleErrorWith(f: (Throwable) -> IOOf<A>): IO<A> =
-  IO.Bind(fix(), IOFrame.Companion.ErrorHandler(f))
+  BIO.Bind(fix(), IOFrame.Companion.ErrorHandler(f))
+
+/**
+ * Executes the given `finalizer` when the source is finished, either in success or in error, or if canceled, allowing
+ * for differentiating between exit conditions. That's thanks to the [ExitCase] argument of the finalizer.
+ *
+ * As best practice, it's not a good idea to release resources via `guaranteeCase` in polymorphic code.
+ * Prefer [bracketCase] for the acquisition and release of resources.
+ *
+ * @see [guarantee] for the simpler version
+ * @see [bracketCase] for the more general operation
+ *
+ */
+fun <A> IOOf<A>.guaranteeCase(finalizer: (ExitCase<Throwable>) -> IOOf<Unit>): IO<A> =
+  IOBracket.guaranteeCase(fix(), finalizer)
+
+/**
+ * Transform the [IO] value of [A] by sequencing an effect [IO] that results in [B].
+ *
+ * @param f function that returns the [IO] effect resulting in [B] based on the input [A].
+ * @returns an effect that results in [B].
+ *
+ * ```kotlin:ank:playground
+ * import arrow.fx.IO
+ *
+ * fun main(args: Array<String>) {
+ *   val result =
+ *   //sampleStart
+ *   IO.just("Hello").flatMap { IO { "$it World" } }
+ *   //sampleEnd
+ *   println(result.unsafeRunSync())
+ * }
+ * ```
+ */
+fun <E, A, B, E2 : E> BIOOf<E, A>.flatMap(f: (A) -> BIOOf<E2, B>): BIO<E2, B> =
```

Comment:
Where's `IO.flatMap` now?

Reply:
`typalias IO<A> = BIO<Nothing, A>`, this is the new `flatMap` that works for both.

Reply:
Oh, I've seen some methods defined over `BIO` and other's over `IO` so that got me confused for a second.